browser/brave_shields/domain_block_page_browsertest.cc

```diff
@@ -299,6 +300,41 @@ IN_PROC_BROWSER_TEST_F(DomainBlockTest, NoFetch) {
   ASSERT_EQ(0, request_count_);
 }
 
+IN_PROC_BROWSER_TEST_F(DomainBlockTest, NoThirdPartyInterstitial) {
+  ASSERT_TRUE(InstallDefaultAdBlockExtension());
+  ASSERT_TRUE(g_brave_browser_process->ad_block_custom_filters_service()
+                  ->UpdateCustomFilters("||b.com^$third-party"));
+
+  GURL url = embedded_test_server()->GetURL("a.com", "/simple_link.html");
+  GURL cross_url =
+      embedded_test_server()->GetURL("a.com", "/cross-site/b.com/simple.html");
+
+  // Navigate to a page on a.com. This should work normally.
+  NavigateTo(url);
+  ASSERT_FALSE(IsShowingInterstitial());
+
+  // Navigate to a page on the third-party b.com. There should be no
+  // interstitial shown.
+  std::string clickLink =
+      "domAutomationController.send(clickLink('" + cross_url.spec() + "'));";
+  bool success = false;
+  EXPECT_TRUE(
+      ExecuteScriptAndExtractBool(web_contents(), clickLink.c_str(), &success));
+  EXPECT_TRUE(success);
+  EXPECT_TRUE(WaitForLoadStop(web_contents()));
+
+  // No interstitial should be shown, since top-level requests are never
+  // third-party.
+  ASSERT_FALSE(IsShowingInterstitial());
+
+  // The default "blocked by an extension" interstitial also should not be
+  // shown. This would appear if the request was blocked by the network delegate
+  // helper.
+  const std::string location =
+      EvalJs(web_contents(), "window.location.href").ExtractString();
+  ASSERT_STRNE("chrome-error://chromewebdata/", location.c_str());
```

Comment:
no need for c_str

Reply:
I think we do need it, when I remove it I get this:

```
../../third_party/googletest/src/googletest/include/gtest/gtest.h:1669:28: note: candidate function not viable: no known conversion from 'const std::string' (aka 'const basic_string<char>') to 'const char *' for 4th argument
GTEST_API_ AssertionResult CmpHelperSTRNE(const char* s1_expression,
                           ^
../../third_party/googletest/src/googletest/include/gtest/gtest.h:1694:28: note: candidate function not viable: no known conversion from 'const char [30]' to 'const wchar_t *' for 3rd argument
GTEST_API_ AssertionResult CmpHelperSTRNE(const char* s1_expression,
```